src/main/java/com/charan/parkingsystem/parking/ParkingLot.java

```diff
@@ -0,0 +1,120 @@
+package com.charan.parkingsystem.parking;
+
+import com.charan.parkingsystem.Types;
+import com.charan.parkingsystem.display.BikeDisplay;
+import com.charan.parkingsystem.display.CarDisplay;
+import com.charan.parkingsystem.display.TruckDisplay;
+import com.charan.parkingsystem.strategy.ParkStrategy;
+import com.charan.parkingsystem.vehicle.Vehicle;
+
+import java.util.*;
+
+public class ParkingLot {
+    private String id = "PR1234";
+    private final int slotsPerFloor;
+    private final int noOfFloors;
+    private Map<Ticket, Vehicle> ticketVehicleMap = new HashMap<>();
+    private List<ParkingFloor> floorList = new ArrayList<>();
+
+    private static ParkingLot instance;
+
+    private ParkingLot(String id, int floors, int slotsPerFloor) {
+        this.id = id;
+        this.noOfFloors = floors;
+        this.slotsPerFloor = slotsPerFloor;
+        this.floorList = setFloorList();
+    }
+
+    private List<ParkingFloor> setFloorList() {
+        for (int i = 1; i <= this.noOfFloors; i++) {
+            List<ParkingSlot> slotsListOnFloor = createSlotList();
+            ParkingFloor floor = new ParkingFloor(i, slotsListOnFloor);
+            floorList.add(floor);
+        }
+        return floorList;
+    }
+
+    private List<ParkingSlot> createSlotList() {
+        List<ParkingSlot> emptySlots = new ArrayList<>();
+        for (int i = 1; i <= this.slotsPerFloor; i++) {
+            ParkingSlot slot = new ParkingSlot(i);
+            emptySlots.add(slot);
+        }
+        return emptySlots;
+    }
+
+    public static ParkingLot getInstance() {
+        if (instance == null) {
+            throw new AssertionError("Call the init first");
```

Comment:
Error messages should be helpful to the user using it. `Call the init first` may not be helpful at all to the user to know what he/she did wrong.

Reply:
Btw, using Singleton for ParkingLot in this case is a very good decision. Well done.